src/switchroot/ostree-mount-util.h

```diff
@@ -40,6 +40,8 @@
 #define _OSTREE_SYSROOT_READONLY_STAMP "/run/ostree-sysroot-ro.stamp"
 #define _OSTREE_COMPOSEFS_ROOT_STAMP "/run/ostree-composefs-root.stamp"
 
+#define autofree_char char __attribute__ ((__cleanup__ (free_char)))
```

Comment:
I don't think you need the _char part here, or in free_char. This can free anything that malloc returned. I would propose just "autofree" define and "free_ptr()".